### PR TITLE
add a simple put saver, for saving to a webdav or REST server

### LINKED ISSUE #738: Add a WebDAV Saver
As discussed in this thread, it turns out to be pretty simple to get WebDAV up and running:

https://groups.google.com/d/topic/tiddlywikidev/eb5JHNm_eak/discussion

@FND has hacked together a working implementation:

https://gist.github.com/FND/b3e969f5ed471b55c831

### PR DESCRIPTION
**Pull request by edrex**
Fixes #738

### CODE REVIEW

**Comment by edrex** [comment]
tested with digest auth

**Comment by Jermolene** [comment]
Hi @edrex thanks for this, looks good. I just realised, though, that you haven't signed the CLA; could you kindly do that in a separate pull request, as described in contributing.md in the root of the repo? Many thanks.

**Comment by edrex** [comment]
signed #2286

File: core/modules/savers/put.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Select the appropriate saver module and set it up
+*/
+var PutSaver = function(wiki) {
+	this.wiki = wiki;
+};
+
+PutSaver.prototype.save = function(text,method,callback) {
+	var req = new XMLHttpRequest();
+	req.addEventListener("load", function() {
+		if (this.status === 200 || this.status === 201) {
```
Comment (Jermolene) [inline core/modules/savers/put.js]: I just realised that this error checking needs completion. We need `callback(null);` if there was no error, and `callback(http.responseText);` if there was a problem.

**Pull request merged**